django/utils/dateparse.py

```diff
@@ -87,12 +89,18 @@ def parse_time(value):
     Return None if the input isn't well formatted, in particular if it
     contains an offset.
     """
-    match = time_re.match(value)
-    if match:
-        kw = match.groupdict()
-        kw['microsecond'] = kw['microsecond'] and kw['microsecond'].ljust(6, '0')
-        kw = {k: int(v) for k, v in kw.items() if v is not None}
-        return datetime.time(**kw)
+    try:
+        # The fromisoformat() method takes timezone info ([+HH:MM[:SS[.ffffff]]])
+        # into account and returns a time with a tzinfo component if possible.
+        # Because the regex fallback discards the same information (it's uncaptured)
+        # we need to remove it here.
```

Comment:
regex is not the main reason to remove the time zone, aware `datetime.time` objects just don't make much sense (see https://github.com/django/django/pull/14583#issuecomment-873395340). I updated this comment.

Reply:
Sorry, when I read your prior comment about tzinfo on a time not making much sense, I thought it was in the context of this & keeping backwards compatibility. I'm fine with whatever works for you comment-wise, given you're more likely to have to re-read it than I :)
I'll be honest, @aaugustin's comments there go far beyond my understanding at this time in the day, but AFAIK aware `datetime.time` objects make perfect sense _in the general abstract sense_ (not in this function, which is intentionally timezone ignorant), and _seems_ (to this lay person) backed up by Python having the concept of aware times. 

_A lot_ of the weirdnesses around timezone handling are _caused_ by the date portion and our historic record (collectively, as a species) of adjusting the offset for things like daylight savings, the separation of a country, adjustment for better interoperability with neighbours, etc. But take all of that adjustment away, and certain things hold true. Hong Kong is always UTC+8, regardless of _time of year_ ... until you get a date portion < 1979, and then there's UTC+8 & UTC+9 to ruin your expectations.
Which is why a time + tznfo is a _fixed offset_ rather than a named timezone. UTC+8 is _always_ 28,000 seconds (ugh, give or take leap seconds and whatnot, probably), but denominated timezones are intrinsically subject to change, and do all the time (hence the necessity of a historical record of them in pytz etc)

Reply:
If I can bring a little bit of nuance to my position.

Yes, Python supports aware time. However, the majority opinion in Django contributions (AFAIK) is that using this feature is likely to result in worse design than not using it.

Many users aren't experts able to delineate narrow sets of circumstances under which code manipulating aware times is more likely to be correct (e.g. "my code will never be used outside HK and HK will never introduce DST [alternative: I will write a unit test that fails if HK ever introduces DST]").

The recommendation would be to manage the time and the timezone in separate objects.

There are other cases where Django diverges from Python's standard behavior. For example, I have found the transaction behavior mandated by PEP 249 less than helpful for most users and I have decided to default to autocommit in Django instead.

Reply:
Thanks Aymeric, I fully understand that stance [that you and Mariusz both shared]; it's a relief to know my understanding of time (or lack thereof) wasn't about to be shaken up completely though :)